### DISCUSSION TITLE
404 when loading signed thumbnail asset blob

### DISCUSSION THREAD

**Discussion by ampledata** [microsoft/PlanetaryComputer#53] [General]
Hi All,

I'm getting the following error when attempting to load a thumbnail asset blob using the `sign()` function:

`HTTPError: HTTP Error 404: The specified blob does not exist.`

If I load the example thumbnail asset, it works fine:

`thumbnail_href = "https://landsateuwest.blob.core.windows.net/landsat-c2/level-2/standard/oli-tirs/2020/047/027/LC08_L2SP_047027_20201204_20210313_02_T1/LC08_L2SP_047027_20201204_20210313_02_T1_thumb_small.jpeg"`

If I load my thumbnail asset, I get a stacktrace:

` thumbnail_href = "https://landsateuwest.blob.core.windows.net/landsat-c2/level-2/standard/oli-tirs/2022/044/034/LC08_L2SP_044034_20220409_20220419_02_T1/LC08_L2SP_044034_20220409_20220419_02_T1_thumb_small.jpeg"`

```
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
Input In [51], in <module>
      1 signed_href = planetary_computer.sign(thumbnail_asset.href)
----> 2 Image.open(urlopen(signed_href))

File /srv/conda/envs/notebook/lib/python3.8/urllib/request.py:222, in urlopen(url, data, timeout, cafile, capath, cadefault, context)
    220 else:
    221     opener = _opener
--> 222 return opener.open(url, data, timeout)

File /srv/conda/envs/notebook/lib/python3.8/urllib/request.py:531, in OpenerDirector.open(self, fullurl, data, timeout)
    529 for processor in self.process_response.get(protocol, []):
    530     meth = getattr(processor, meth_name)
--> 531     response = meth(req, response)
    533 return response

File /srv/conda/envs/notebook/lib/python3.8/urllib/request.py:640, in HTTPErrorProcessor.http_response(self, request, response)
    637 # According to RFC 2616, "2xx" code indicates that the client's
    638 # request was successfully received, understood, and accepted.
    639 if not (200 <= code < 300):
--> 640     response = self.parent.error(
    641         'http', request, response, code, msg, hdrs)
    643 return response

File /srv/conda/envs/notebook/lib/python3.8/urllib/request.py:569, in OpenerDirector.error(self, proto, *args)
    567 if http_err:
    568     args = (dict, 'default', 'http_error_default') + orig_args
--> 569     return self._call_chain(*args)

File /srv/conda/envs/notebook/lib/python3.8/urllib/request.py:502, in OpenerDirector._call_chain(self, chain, kind, meth_name, *args)
    500 for handler in handlers:
    501     func = getattr(handler, meth_name)
--> 502     result = func(*args)
    503     if result is not None:
    504         return result

File /srv/conda/envs/notebook/lib/python3.8/urllib/request.py:649, in HTTPDefaultErrorHandler.http_error_default(self, req, fp, code, msg, hdrs)
    648 def http_error_default(self, req, fp, code, msg, hdrs):
--> 649     raise HTTPError(req.full_url, code, msg, hdrs, fp)

HTTPError: HTTP Error 404: The specified blob does not exist.
```

I'm not sure if sharing works, but here's my notebook: https://pccompute.westeurope.cloudapp.azure.com/compute/hub/user-redirect/lab/tree/Bolinas.ipynb

![Screen Shot 2022-05-02 at 6 15 48 PM](https://user-images.githubusercontent.com/91230/166391343-57385a78-b71c-4db2-a308-92d704b6f52a.png)

Thanks!

**Comment by ampledata** [comment]
The `thumbnail_asset` dict for reference:

```
{
  "href": "https://landsateuwest.blob.core.windows.net/landsat-c2/level-2/standard/oli-tirs/2022/044/034/LC08_L2SP_044034_20220409_20220419_02_T1/LC08_L2SP_044034_20220409_20220419_02_T1_thumb_small.jpeg",
  "type": "image/jpeg",
  "title": "Thumbnail image"
}

```

**Comment by TomAugspurger** [comment]
Just to confirm, https://planetarycomputer.microsoft.com/api/stac/v1/collections/landsat-8-c2-l2/items/LC08_L2SP_044034_20220409_02_T1 is the STAC item you're using?

That error message is saying that the specified blob doesn't exist. And I've just confirmed that it indeed doesn't exist, so something must have broken in our pipeline. I'll check on it and will post an update here.

**Comment by ampledata** [comment]
Aye:
`item_url = "https://planetarycomputer.microsoft.com/api/stac/v1/collections/landsat-8-c2-l2/items/LC08_L2SP_044034_20220409_02_T1"`

**Comment by TomAugspurger** [comment]
OK, here's the situation as I understand it: USGS stopped producing those thumbnails in 2021, so they won't exist for any scenes newer than 2021. Our process for creating STAC items incorrectly assumes that those files are there and generates an asset for them, even when they don't. That's a problem with our STAC items.

That said, we'll soon be deprecating the `landsat-8-c2-l2` collection in favor of the `landsat-c2-l2` collection (which includes data from Landsat 9). This actually fixed this issue with incorrectly adding the `thumbnail` link. 

In the meantime, I would recommend using the `rendered_preview` asset, which uses our data API to dynamically generate a preview from the COG. For example, https://planetarycomputer.microsoft.com/api/data/v1/item/preview.png?collection=landsat-8-c2-l2&item=LC08_L2SP_044034_20220409_02_T1&assets=SR_B4&assets=SR_B3&assets=SR_B2&color_formula=gamma+RGB+2.7%2C+saturation+1.5%2C+sigmoidal+RGB+15+0.55. That will work with both the `landsat-8-c2-l2` and `landsat-c2-l2` collections.

I'll update the example notebook to use the `rendered_preview` asset.

**Comment by ampledata** [comment]
![Screen Shot 2022-05-03 at 4 06 41 PM](https://user-images.githubusercontent.com/91230/166587197-8dbd2346-5d78-42ca-a1c9-c27abd13edbd.png)
Sweet. Thanks!
